x-pack/plugins/security/server/routes/authorization/roles/model/put_payload.ts

```diff
@@ -302,3 +303,50 @@ const transformPrivilegesToElasticsearchPrivileges = (
     };
   });
 };
+
+export const validateKibanaPrivileges = (
+  kibanaFeatures: KibanaFeature[],
+  kibanaPrivileges: PutPayloadSchemaType['kibana']
+) => {
+  const validationErrors = (kibanaPrivileges ?? []).flatMap((priv) => {
+    const forAllSpaces = priv.spaces.includes(ALL_SPACES_ID);
+
+    return Object.entries(priv.feature ?? {}).flatMap(([featureId, feature]) => {
+      const errors: string[] = [];
+      const kibanaFeature = kibanaFeatures.find((f) => f.id === featureId);
+      if (!kibanaFeature) return errors;
+
+      if (feature.includes('all')) {
+        if (kibanaFeature.privileges?.all.disabled) {
+          errors.push(`Feature [${featureId}.all] is disabled and cannot be specified`);
```

Comment:
I think a more pedantically correct error message would be something like:

```suggestion
          errors.push(`Feature [${featureId}] does not support privilege [all].`);
```

The distinction being that `${featureId}.all` is not a feature, but rather a privilege.